pkg/resource/deploy/step.go

```diff
@@ -237,6 +237,8 @@ func NewDeleteReplacementStep(plan *Plan, old *resource.State, pendingDelete boo
 	contract.Assert(old.ID != "" || !old.Custom)
 	contract.Assert(!old.Custom || old.Provider != "" || providers.IsProviderType(old.Type))
 	contract.Assert(!pendingDelete || old.Delete)
+
+	old.PendingReplacement = !pendingDelete
```

Comment:
Perhaps a comment here?  I can't quite piece together in my had why this is needed?

Reply:
+1 to the comment, but in english my understanding is "If we are doing a DeleteReplacement but we are not pending the deletion to the end of the plan, then we are deleting it eagerly as part of a DBR."

Reply:
Yep. I will add a comment.